python/src/opendp/context.py

```diff
@@ -403,6 +403,8 @@ def compositor(
 
         if margins:
             for by, margin in margins.items():
+                # wrap single strings in a singleton tuple
+                by = (by,) if isinstance(by, str) else by
```

Comment:
Possibly steer users towards the tuple syntax?
```suggestion
                if isinstance(by, str):
                    from warnings import warn
                    warn("Tuples preferred for margin keys. For single-valued tuples, include a trailing comma, ie: ('my-key',)")
                    by = (by,)
```

Reply:
I changed the PR to raise an exception if the input is not a tuple, and to use the grouping key in the error message if it is a string.